app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -79,10 +89,32 @@ def google_oauth2
     end
   end
 
-  private def update_user_with_github_attributes(user, login, token)
+private
+
+  def update_user_with_github_attributes(user, login, token)
     user.update_attributes(
       github_login:       login,
       github_oauth_token: token
     )
   end
+
+  def github_get_user_email(client)
+    email = nil
+    begin
+      email = client.emails.select(&:primary).first
+      return email.email unless email.nil?
+
+      email = client.emails.first
+      return email.email unless email.nil?
+    rescue Octokit::ClientError => e
+      Rails.logger.warn "Octokit:ClientError exception while retrieving user's emails. We probably lack user:email permission. Will try to extract email from user's public profile. Error message: #{e}"
+    end
+
+    # Try to get email from public profile
+    if !env["omniauth.auth"].extra.raw_info.email.nil? && env["omniauth.auth"].extra.raw_info.email != ""
```

Comment:
`env["omniauth.auth"].extra.raw_info.email.present?` should do the same

Reply:
Agree! fixed